Schema: "IllegalReferenceCountException: refCnt: 0" while getting schema with AutoConsumeSchema

**Describe the bug**
I am observing this error while working on new tests cases about AutoConsumeSchema.
I am still investigating in order to find a simple way to reproduce the problem, I am not able to reproduce it manually.


```
persistent://public/default/sink-created-test-kv-sink-input-kvavro-yyhymzgt] failed to get schema : io.netty.util.IllegalReferenceCountException: refCnt: 0
java.util.concurrent.CompletionException: io.netty.util.IllegalReferenceCountException: refCnt: 0
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273) ~[?:1.8.0_282]
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:1005) ~[?:1.8.0_282]
	at java.util.concurrent.CompletableFuture.thenCompose(CompletableFuture.java:2137) ~[?:1.8.0_282]
	at org.apache.pulsar.client.impl.ClientCnx.sendGetSchema(ClientCnx.java:879) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.BinaryProtoLookupService.lambda$getSchema$8(BinaryProtoLookupService.java:227) ~[java-instance.jar:?]
	at java.util.concurrent.CompletableFuture.uniAccept(CompletableFuture.java:670) ~[?:1.8.0_282]
	at java.util.concurrent.CompletableFuture.uniAcceptStage(CompletableFuture.java:683) ~[?:1.8.0_282]
	at java.util.concurrent.CompletableFuture.thenAccept(CompletableFuture.java:2010) ~[?:1.8.0_282]
	at org.apache.pulsar.client.impl.BinaryProtoLookupService.getSchema(BinaryProtoLookupService.java:223) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionSchemaInfoProvider.loadSchema(MultiVersionSchemaInfoProvider.java:97) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionSchemaInfoProvider.access$000(MultiVersionSchemaInfoProvider.java:41) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionSchemaInfoProvider$1.load(MultiVersionSchemaInfoProvider.java:54) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionSchemaInfoProvider$1.load(MultiVersionSchemaInfoProvider.java:51) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3951) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4935) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionSchemaInfoProvider.getSchemaByVersion(MultiVersionSchemaInfoProvider.java:75) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.KeyValueSchema$2.getSchemaByVersion(KeyValueSchema.java:221) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.reader.AbstractMultiVersionReader.getSchemaInfoByVersion(AbstractMultiVersionReader.java:115) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.generic.MultiVersionGenericAvroReader.loadReader(MultiVersionGenericAvroReader.java:43) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.reader.AbstractMultiVersionReader$1.load(AbstractMultiVersionReader.java:52) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.reader.AbstractMultiVersionReader$1.load(AbstractMultiVersionReader.java:49) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$LoadingValueReference.loadFuture(LocalCache.java:3529) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.loadSync(LocalCache.java:2278) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.lockedGetOrLoad(LocalCache.java:2155) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$Segment.get(LocalCache.java:2045) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache.get(LocalCache.java:3951) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache.getOrLoad(LocalCache.java:3974) ~[java-instance.jar:?]
	at com.google.common.cache.LocalCache$LocalLoadingCache.get(LocalCache.java:4935) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.reader.AbstractMultiVersionReader.read(AbstractMultiVersionReader.java:86) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.AbstractStructSchema.decode(AbstractStructSchema.java:60) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.KeyValueSchema.decode(KeyValueSchema.java:160) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.KeyValueSchema.lambda$decode$0(KeyValueSchema.java:151) ~[java-instance.jar:?]
	at org.apache.pulsar.common.schema.KeyValue.decode(KeyValue.java:146) [java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.KeyValueSchema.decode(KeyValueSchema.java:151) [java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.KeyValueSchema.decode(KeyValueSchema.java:37) [java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.decode(AutoConsumeSchema.java:103) [java-instance.jar:?]
	at org.apache.pulsar.client.impl.schema.AutoConsumeSchema.decode(AutoConsumeSchema.java:41) [java-instance.jar:?]
	at org.apache.pulsar.client.impl.MessageImpl.getValue(MessageImpl.java:356) [java-instance.jar:?]
	at org.apache.pulsar.functions.source.PulsarRecord.getValue(PulsarRecord.java:81) [org.apache.pulsar-pulsar-functions-instance-2.8.0-SNAPSHOT.jar:2.8.0-SNAPSHOT]
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.readInput(JavaInstanceRunnable.java:375) [org.apache.pulsar-pulsar-functions-instance-2.8.0-SNAPSHOT.jar:?]
	at org.apache.pulsar.functions.instance.JavaInstanceRunnable.run(JavaInstanceRunnable.java:239) [org.apache.pulsar-pulsar-functions-instance-2.8.0-SNAPSHOT.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_282]
Caused by: io.netty.util.IllegalReferenceCountException: refCnt: 0
	at io.netty.buffer.AbstractByteBuf.ensureAccessible(AbstractByteBuf.java:1454) ~[java-instance.jar:?]
	at io.netty.buffer.AbstractByteBuf.checkIndex(AbstractByteBuf.java:1383) ~[java-instance.jar:?]
	at io.netty.buffer.UnsafeByteBufUtil.getBytes(UnsafeByteBufUtil.java:481) ~[java-instance.jar:?]
	at io.netty.buffer.PooledUnsafeDirectByteBuf.getBytes(PooledUnsafeDirectByteBuf.java:130) ~[java-instance.jar:?]
	at io.netty.buffer.PooledSlicedByteBuf.getBytes(PooledSlicedByteBuf.java:235) ~[java-instance.jar:?]
	at io.netty.buffer.ByteBufUtil.decodeString(ByteBufUtil.java:1147) ~[java-instance.jar:?]
	at io.netty.buffer.AbstractByteBuf.toString(AbstractByteBuf.java:1246) ~[java-instance.jar:?]
	at org.apache.pulsar.common.api.proto.LightProtoCodec.readString(LightProtoCodec.java:250) ~[java-instance.jar:?]
	at org.apache.pulsar.common.api.proto.Schema.getName(Schema.java:102) ~[java-instance.jar:?]
	at org.apache.pulsar.common.protocol.schema.SchemaInfoUtil.newSchemaInfo(SchemaInfoUtil.java:48) ~[java-instance.jar:?]
	at org.apache.pulsar.client.impl.ClientCnx.lambda$sendGetSchema$12(ClientCnx.java:891) ~[java-instance.jar:?]
	at java.util.concurrent.CompletableFuture.uniComposeStage(CompletableFuture.java:995) ~[?:1.8.0_282]
	... 45 more
```

--- a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java
+++ b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java
@@ -487,7 +487,7 @@ public class ClientCnx extends PulsarHandler {
         CompletableFuture<CommandGetLastMessageIdResponse> requestFuture =
                 (CompletableFuture<CommandGetLastMessageIdResponse>) pendingRequests.remove(requestId);
         if (requestFuture != null) {
-            requestFuture.complete(success);
+            requestFuture.complete(new CommandGetLastMessageIdResponse().copyFrom(success));
         } else {
             log.warn("{} Received unknown request id from server: {}", ctx.channel(), success.getRequestId());
         }
@@ -803,7 +803,7 @@ public class ClientCnx extends PulsarHandler {
             log.warn("{} Received unknown request id from server: {}", ctx.channel(), requestId);
             return;
         }
-        future.complete(commandGetSchemaResponse);
+        future.complete(new CommandGetSchemaResponse().copyFrom(commandGetSchemaResponse));
     }
 
     @Override
@@ -816,7 +816,7 @@ public class ClientCnx extends PulsarHandler {
             log.warn("{} Received unknown request id from server: {}", ctx.channel(), requestId);
             return;
         }
-        future.complete(commandGetOrCreateSchemaResponse);
+        future.complete(new CommandGetOrCreateSchemaResponse().copyFrom(commandGetOrCreateSchemaResponse));
     }
 
     Promise<Void> newPromise() {

